File: pilha/pilha.js
var Stack = /** @class */ (function () {
    function Stack(maxStackSize) {
        this.maxStackSize = maxStackSize;
        this.stack = [];
    }
    Stack.prototype.getStack = function () {
        return this.stack;
    };
    Stack.prototype.getStackLength = function () {
        return this.stack.length;
    };
    Stack.prototype.stackIsEmpty = function () {
        return this.stack.length === 0;
    };
    Stack.prototype.stackIsFull = function () {
        return this.stack.length === this.maxStackSize;
    };
    Stack.prototype.push = function (value) {
        if (!this.stackIsFull())
            this.stack.push(value);
        else
            throw Error('Stack is full');
    };
    Stack.prototype.pop = function () {
        if (!this.stackIsEmpty())
            return this.stack.pop();
        else
            throw Error('Stack is empty');
    };
    Stack.prototype.top = function () {
        if (!this.stackIsEmpty())
            return this.stack[this.getStackLength() - 1];
        else
            throw Error('Stack is empty');
    };
    return Stack;
}());
var stackExample = new Stack(5);
console.log(stackExample.getStackLength()); // 0
console.log(stackExample.stackIsEmpty()); // true
console.log(stackExample.stackIsFull()); // false
stackExample.push(1);
stackExample.push('banana');
console.log(stackExample.getStack()); // [1, 'banana']
console.log(stackExample.pop()); // 'banana'
console.log(stackExample.getStack()); // [1]
console.log(stackExample.top()); // 1
console.log(stackExample.pop()); // 1
console.log(stackExample.getStack()); // []

File: pilha/pilha.py
class Stack:
    def __init__(self, maxStackSize):
        self.__stack = []
        self.__maxStackSize = maxStackSize

    def getStack(self):
        return self.__stack

    def getStackLength(self):
        return len(self.__stack)

    def stackIsEmpty(self):
        return self.getStackLength() == 0

    def stackIsFull(self):
        return self.getStackLength() == self.__maxStackSize

    def push(self, element):
        if (not self.stackIsFull()):
            self.__stack.append(element)
        else:
            raise Exception('Stack is full')

    def pop(self):
        if (not self.stackIsEmpty()):
            return self.__stack.pop()
        else:
            raise Exception('Stack is empty')

    def top(self):
        if (not self.stackIsEmpty()):
            return self.__stack[self.getStackLength() - 1]
        else:
            raise Exception('Stack is empty')


stackExample = Stack(4)
# The commented code below will raise a "Stack is empty" Exception
# stackExample.top()
stackExample.push(1)
print('Executed Push with value 1')
print('Stack top: ' + str(stackExample.top()))
stackExample.push(2) 
print('Executed Push with value 2')
print('Stack top: ' + str(stackExample.top()))
stackExample.push(3) 
print('Executed Push with value 3')
print('Stack top: ' + str(stackExample.top()))
stackExample.push(4)
print('Executed Push with value 4')
print('Stack top: ' + str(stackExample.top()))
# The commented code below will raise a "Stack is full" Exception
# stackExample.push(5)
stackExample.pop()
print('Executed Pop')
print('Stack top: ' + str(stackExample.top()))
stackExample.pop()
print('Executed Pop')
print('Stack top: ' + str(stackExample.top()))
stackExample.pop()
print('Executed Pop')
print('Stack top: ' + str(stackExample.top()))
# The commented code below will raise a "Stack is empty" Exception
# stackExample.pop()